lib/AST/ASTContext.cpp

```diff
@@ -4074,6 +4074,31 @@ LayoutConstraint LayoutConstraint::getLayoutConstraint(LayoutConstraintKind Kind
                                                       unsigned SizeInBits,
                                                       unsigned Alignment,
                                                       ASTContext &C) {
+  if (!LayoutConstraintInfo::isKnownSizeTrivial(Kind)) {
+    assert(SizeInBits == 0);
+    assert(Alignment == 0);
+
+    switch(Kind) {
+    case LayoutConstraintKind::Trivial:
+      return LayoutConstraint(&LayoutConstraintInfo::TrivialConstraintInfo);
+    case LayoutConstraintKind::NativeClass:
+      return LayoutConstraint(&LayoutConstraintInfo::NativeClassConstraintInfo);
+    case LayoutConstraintKind::Class:
+      return LayoutConstraint(&LayoutConstraintInfo::ClassConstraintInfo);
+    case LayoutConstraintKind::NativeRefCountedObject:
+      return LayoutConstraint(
+          &LayoutConstraintInfo::NativeRefCountedObjectConstraintInfo);
+    case LayoutConstraintKind::RefCountedObject:
+      return LayoutConstraint(
+          &LayoutConstraintInfo::RefCountedObjectConstraintInfo);
+    case LayoutConstraintKind::UnknownLayout:
+      return LayoutConstraint(
+          &LayoutConstraintInfo::UnknownLayoutConstraintInfo);
+    default:
```

Comment:
We don't need the default, right? It'll end up suppressing warnings that we'd rather see when the switch isn't covering all of the cases.